k8sdeps/transformer/patch/conflictdetector.go

```diff
@@ -123,19 +124,25 @@ func (smp *strategicMergePatch) findConflict(
 }
 
 func (smp *strategicMergePatch) mergePatches(patch1, patch2 *resource.Resource) (*resource.Resource, error) {
+	if hasDeleteDirectiveMarker(patch2.Map()) {
```

Comment:
I don't think this work if the patch with is $patch: delete is third in the list. The entire list of patches needs to be merged up front.

Reply:
Since patches are being merged two at a time the relative order in the greater list of patches shouldn't matter. As long as the delete patch is always the first patch passed to `strategicpatch.MergeStrategicMergeMapPatchUsingLookupPatchMeta`. This prevents the apimachinery logic from processing the directives leaving them intact in the final merged patch.

Reply:
You are probably right. This avoid to much changes. A user that would want to remove more than one key would just have to concentrate all the $patch: delete in one patch

Reply:
Yes exactly, made sure to include that requirement in the documentation update.